src/modules/qt/filter_typewriter.cpp

```diff
@@ -0,0 +1,306 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt.h>
+#include <framework/mlt_filter.h>
+#include <framework/mlt_frame.h>
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <vector>
+
+#include "typewriter.h"
+
+struct producer_ktitle_s
+{
+    struct mlt_producer_s parent;
+    uint8_t *rgba_image;
+    uint8_t *current_image;
+    uint8_t *current_alpha;
+    mlt_image_format format;
+    int current_width;
+    int current_height;
+    int has_alpha;
+    pthread_mutex_t mutex;
+};
+
+typedef struct producer_ktitle_s *producer_ktitle;
+
+struct FilterContainer{
+    XmlParser xp;
+
+    std::vector<TypeWriter> renders;      // rendered data [array]
+    bool init;               // 1 if initialized
+
+    int current_frame;      // currently parsed frame
+
+    std::string xml_data;   // data field content (xml data)
+    bool is_template;
+    int step_length;        // frame step value
+    float sigma;            // sigma of fluctuations
+    int seed;               // seed for random fluctuations
+    int macro;              // macro type: 0 - custom, 1 - char, 2 - word, 3 - line
+
+    int producer_type;      // 1 - kdenlivetitle
+    mlt_producer producer;  // hold producer pointer
+
+    FilterContainer() {
+        clean();
+    }
+
+    void clean()
+    {
+        renders.clear();
+        init = false;
+        current_frame = -1;
+        xml_data.clear();
+        is_template = false;
+        step_length = 0;
+        sigma = 0;
+        seed = 0;
+        macro = 0;
+        producer_type = 0;
+        producer = nullptr;
+    }
+};
+
+/*
+ * Get data for display.
+ */
+static int get_producer_data(mlt_properties filter_p, mlt_properties frame_p, FilterContainer * cont)
+{
+    if (cont == nullptr)
+        return 0;
+
+    char * d = nullptr;
+    int step_length = 0;
+    int sigma = 0;
+    int seed = 0;
+    int macro = 0;
+
+    mlt_producer producer = nullptr;
+    mlt_properties producer_properties = nullptr;
+
+    unsigned int update_mask = 0;
+
+    /* Try with kdenlivetitle */
+    producer_ktitle kt = static_cast<producer_ktitle>(mlt_properties_get_data( frame_p, "producer_kdenlivetitle", NULL ) );
+    if (kt != nullptr)
+    {
+        /* Obtain properties of producer */
+        producer = &kt->parent;
+        producer_properties = MLT_PRODUCER_PROPERTIES( producer );
+
+        if (producer == nullptr || producer_properties == nullptr)
+            return 0;
+
+        d = mlt_properties_get( producer_properties, "resource" );
+        cont->is_template = (d && d[0] != '\0');
+
+        if (cont->is_template)
+            d = mlt_properties_get( producer_properties, "_xmldata");
+        else
+            d = mlt_properties_get( producer_properties, "xmldata" );
+
+        if (d == nullptr)
+            return 0;
+
+        step_length = atoi(mlt_properties_get(filter_p, "step_length"));
```

Comment:
This is the wrong way to convert a property to integer or float. These will fail on null pointers when these properties have not been set or cleared. See `mlt_properties.h` for the ways to get these as integer and double. You must use our functions instead of converting string to numeric like this.

Reply:
Fixed in the new commit.